src/Components/EnvironmentPicker/EnvironmentPickerManager.ts

```diff
@@ -13,10 +13,15 @@ import {
 export const getEnvironmentDisplayText = (env: string | IAzureResource) => {
     try {
         if (env) {
-            const urlObj = new URL(
-                getResourceUrl(env, AzureResourceTypes.DigitalTwinInstance)
-            );
-            return urlObj.hostname.split('.')[0];
+            if (typeof env === 'string') {
+                if (new URL(env)) {
```

Comment:
I know in the past we have seen many exceptions thrown by this URL class. Do we have sufficient safeguards in place now to handle those conditions gracefully?

Reply:
yes, now in the codebase whenever we use URL class it is wrapped in try/catch block and should be handled gracefully.